t/t5324-split-commit-graph.sh

```diff
@@ -425,4 +425,54 @@ done <<\EOF
 0600 -r--------
 EOF
 
+test_expect_success 'setup repo for mixed generation commit-graph-chain' '
```

Comment:
@derrickstolee @jnareb @ttaylorr

While writing test for split commit-graph, I wondered about the following scenario:

We start off with New Git and create a base commit-graph. Then we switch to old Git (possibly installed with an IDE, for example) and add some more commits. If we try to write to a new commit-graph file, we wind up with a chain of commit-graph files with and without GDAT chunks.

In such a case, I have disabled generation numbers and print out a warning.

We could have tried to salvage some of the performance boost by using topological level. I am thinking of setting up adding another flag to struct commit_graph to indicate whether to read topological level or corrected commit date. Will that be worth the trouble?

Reply:
When we are writing a split commit-graph, we should not add a GDAT chunk if the base commit graph doesn't have one. This will prevent "holes" in the middle of a chain, as you described. It means we won't get a GDAT chunk on top of old commit-graphs unless we merge all of the levels.

With this in mind, it would be good to write a commit-graph without a GDAT chunk (using `GIT_TEST_COMMIT_GRAPH_NO_GDAT=1`), then add a new layer on top (without that env var) and see that the GDAT chunk doesn't exist there. Finally, rewrite the whole commit-graph using `--split=replace` and see that the single layer _does_ have a GDAT chunk.

Does that make sense?

Reply:
I agree with @derrickstolee that when writing section of a split commit-graph we should not add a GDAT chunk if the previous file in the chain doesn't have it -- this solution simplifies reasoning (and should be only temporary measure until all Git is upgraded to generation number v2 capable version).

I'm not sure if another flag in `struct commit_graph` would be necessary, though it might be the best solution anyway. If the topmost commit we start from, or the top file in the commit-graph chain doesn't have generation number v2 (no GDAT chunk), we switch to generation number v1. Otherwise we can safely use generation number v2.

Reply:
> When we are writing a split commit-graph, we should not add a GDAT chunk if the base commit graph doesn't have one. This will prevent "holes" in the middle of a chain, as you described. It means we won't get a GDAT chunk on top of old commit-graphs unless we merge all of the levels.
> 
> With this in mind, it would be good to write a commit-graph without a GDAT chunk (using `GIT_TEST_COMMIT_GRAPH_NO_GDAT=1`), then add a new layer on top (without that env var) and see that the GDAT chunk doesn't exist there. Finally, rewrite the whole commit-graph using `--split=replace` and see that the single layer _does_ have a GDAT chunk.
> 
> Does that make sense?

Yes, that makes complete sense to me. I can see why we shouldn't merge split commit-graph chain unless explicitly ordered by the user.